lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationService.java

```diff
@@ -54,39 +58,34 @@ public IdentityVerificationService(
 
     public DocumentCheckVerificationResult verifyIdentity(
             DrivingPermitForm drivingPermitData, ThirdPartyAPIService thirdPartyAPIService)
-            throws OAuthHttpResponseExceptionWithErrorBody {
+            throws OAuthErrorResponseException {
         DocumentCheckVerificationResult result = new DocumentCheckVerificationResult();
 
         try {
             LOGGER.info("Validating form data...");
             ValidationResult<List<String>> validationResult =
                     this.formDataValidator.validate(drivingPermitData);
+
+            IssuingAuthority issuingAuthority =
```

Comment:
Should we be reading this before its had the validation status checked?

Reply:
This is mapping the issuer in the form data to an enum, just for logging the issuer + capturing the issuer metric.

Previously the case was that the form validation could be marked as valid. Then when logging the issuer it could fail if not matching the enum.

Since this is still confusing I will add an enum mapping check to the form validator as this is a kind of validation. Then move the logging and metric capture after the form validation.

```
try {
    IssuingAuthority.valueOf(drivingPermitForm.getLicenceIssuer());
}  catch (IllegalArgumentException e) {
    validationErrors.add(String.format("LicenceIssuer %s is Unknown", drivingPermitForm.getLicenceIssuer()));
}
```